### PR TITLE
feat(Dispatcher): Default `UserFriendlyException` does not support retries

### PR DESCRIPTION
**Pull request by zhenlei520**
# Description

Default `UserFriendlyException` does not support retries

## Checklist

Please make sure you've completed the relevant tasks for this PR, out of the following list:

* [x] Code compiles correctly
* [x] Created/updated tests
* [x] Extended the documentation

### CODE REVIEW

File: src/Dispatcher/Masa.Contrib.Dispatcher.Events/EventBusBuilder.cs [hunk 1]
```diff
@@ -3,10 +3,16 @@
 
 namespace Masa.Contrib.Dispatcher.Events;
 
-public class EventBusBuilder: IEventBusBuilder
+public class EventBusBuilder : IEventBusBuilder
 {
     public IServiceCollection Services { get; }
 
+    /// <summary>
+    /// Used for logging in default retry policy
```
Comment (doddgu) [inline src/Dispatcher/Masa.Contrib.Dispatcher.Events/EventBusBuilder.cs:11]: 用于登录默认重试策略???

File: src/Dispatcher/Masa.Contrib.Dispatcher.Events/IStrategyExceptionProvider.cs [hunk 2]
```diff
@@ -0,0 +1,11 @@
+﻿// Copyright (c) MASA Stack All rights reserved.
+// Licensed under the MIT License. See LICENSE.txt in the project root for license information.
+
+namespace Masa.Contrib.Dispatcher.Events;
+
+public interface IStrategyExceptionProvider
```
Comment (doddgu) [inline src/Dispatcher/Masa.Contrib.Dispatcher.Events/IStrategyExceptionProvider.cs:6]: StrategyException -> ExceptionStrategy?

**Review by doddgu — CHANGES_REQUESTED** [review]

File: src/Dispatcher/Masa.Contrib.Dispatcher.Events/EventBusBuilder.cs (hunk 1 above)
Comment (zhenlei520) [reply src/Dispatcher/Masa.Contrib.Dispatcher.Events/EventBusBuilder.cs:11]: 已弃用的属性，已删除

File: src/Dispatcher/Masa.Contrib.Dispatcher.Events/IStrategyExceptionProvider.cs (hunk 2 above)
Comment (zhenlei520) [reply src/Dispatcher/Masa.Contrib.Dispatcher.Events/IStrategyExceptionProvider.cs:6]: Processed

**Review by doddgu — APPROVED** [review]
LGTM

**Pull request merged**